### PR TITLE
Bump k8s to 1.14.1

### PR DESCRIPTION
**Pull request by ingvagabund**
- bump k8s to v1.14.1
- Manager.GetRecorder renamed to Manager.GetEventRecorderFor
- replace "-" with "_" in controller names (the name is used to prefix prometheus metric name which refuses "-" in metric name, see https://github.com/kubernetes-sigs/controller-runtime/pull/376)
  - machineset-controller -> machineset_controller
  - machinedeployment-controller -> machinedeployment_controller
  - cluster-controller -> cluster_controller
- replace order of arguments in Client.List (runtime object switched with list options, list options converted to a function)
- schema moved from "sigs.k8s.io/controller-runtime/pkg/runtime/schema" under "sigs.k8s.io/controller-runtime/pkg/scheme"
- "k8s.io/client-go/util/integer" moved under "k8s.io/utils/integer"

### CODE REVIEW

File: pkg/controller/machinedeployment/machinedeployment_controller_test.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ func TestReconcile(t *testing.T) {
 	// Verify that the MachineSet was created.
 	machineSets := &machinev1beta1.MachineSetList{}
 	expectInt(t, 1, func(ctx context.Context) int {
-		if err := c.List(ctx, &client.ListOptions{}, machineSets); err != nil {
+		if err := c.List(ctx, machineSets, func(opts *client.ListOptions) {}); err != nil {
```
Comment (bison) [inline pkg/controller/machinedeployment/machinedeployment_controller_test.go]: What's up with the empty functions? Can you not pass `nil`?

Comment (ingvagabund) [reply pkg/controller/machinedeployment/machinedeployment_controller_test.go]: It panics unfortunately. Already tried.

File: pkg/controller/machinedeployment/controller.go [hunk 2]
```diff
@@ -108,9 +108,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler, mapFn handler.ToRequestsFu
 func (r *ReconcileMachineDeployment) getMachineSetsForDeployment(d *v1beta1.MachineDeployment) ([]*v1beta1.MachineSet, error) {
 	// List all MachineSets to find those we own but that no longer match our selector.
 	machineSets := &v1beta1.MachineSetList{}
-	listOptions := &client.ListOptions{Namespace: d.Namespace}
-
-	if err := r.Client.List(context.Background(), listOptions, machineSets); err != nil {
+	if err := r.Client.List(context.Background(), machineSets, func(opt *client.ListOptions) { opt.InNamespace(d.Namespace) }); err != nil {
```
Comment (bison) [inline pkg/controller/machinedeployment/controller.go]: This looks weird. Aren't the options here a variadic parameter, and they give you a bunch of functions that return `ListOptionFunc`, so you can do something like what's below?

`r.Client.List(context.Background(), machineSets, client.InNamespace(ns), client.MatchingLabels(labels))`

File: pkg/controller/machinedeployment/machinedeployment_controller_test.go (hunk 1 above)
Comment (bison) [reply pkg/controller/machinedeployment/machinedeployment_controller_test.go]: What is the panic? That seems like a bug for sure. We should at least open an issue about that and link here or something.

Comment (bison) [reply pkg/controller/machinedeployment/machinedeployment_controller_test.go]: Also, I guess you shouldn't even need `nil`. You should be able to just not pass this parameter.

File: pkg/controller/machinedeployment/controller.go (hunk 2 above)
Comment (ingvagabund) [reply pkg/controller/machinedeployment/controller.go]: Good point. PR updated.

File: pkg/controller/machine/controller.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ func AddWithActuator(mgr manager.Manager, actuator Actuator) error {
 func newReconciler(mgr manager.Manager, actuator Actuator) reconcile.Reconciler {
 	r := &ReconcileMachine{
 		Client:        mgr.GetClient(),
-		eventRecorder: mgr.GetRecorder("machine-controller"),
+		eventRecorder: mgr.GetEventRecorderFor("machine-controller"),
```
Comment (frobware) [inline pkg/controller/machine/controller.go:64]: Should this also be s/-/_?

Comment (ingvagabund) [reply pkg/controller/machine/controller.go:64]: I would still prefer to keep the event source to have `-` instead of `_`.

**Pull request merged**